openassessment/xblock/static/js/src/studio/oa_edit_settings.js

```diff
@@ -27,11 +28,38 @@ OpenAssessment.EditSettingsView = function(element, assessmentViews) {
         "#openassessment_submission_due_time"
     ).install();
 
+    new OpenAssessment.SelectControl(
+        $("#openassessment_submission_upload_selector", this.element),
+        {'custom': $("#openassessment_submission_white_listed_file_types_wrapper", this.element)},
+        new OpenAssessment.Notifier([
+            new OpenAssessment.AssessmentToggleListener()
+        ])
+    ).install();
+
     this.leaderboardIntField = new OpenAssessment.IntField(
         $("#openassessment_leaderboard_editor", this.element),
         { min: 0, max: 100 }
     );
 
+    this.fileTypeWhiteListInputField = new OpenAssessment.InputControl(
+        $("#openassessment_submission_white_listed_file_types", this.element),
+        function(value) {
+            var badExts = [];
+            var errors = [];
+            var whiteList = $.map(value.split(','), $.trim);
+            $.each(whiteList, function(index, ext) {
+                if (data.FILE_EXT_BLACK_LIST.indexOf(ext) !== -1) {
+                    badExts.push(ext);
+                }
+            });
+            if (badExts.length) {
+                errors.push(gettext('The following file types are not allowed: ') + badExts.join(','));
+            }
```

Comment:
I just tried entering "dmg, .exe" in the sandbox, and was surprised that the error only listed the dmg file as not allowed. It looks as though the values entered have to match the black list exactly, so you can't enter an extension with capital letters or punctuation and have the check against the black list catch them.
A couple of thoughts: 
- is the edit check on the files that students upload case sensitive? that is, if someone tries to upload "really_bad.dmg" they wouldn't be allowed to, and if they try to upload "really_bad.DMG" they still wouldn't be allowed to, correct?
- should the blacklist contain the all-upper-case versions of these file extensions as well as the all lower case versions?
- should the blacklist contain the versions of these file extensions that include a period?
(The blacklist might then grow to include four variations for each file type: dmg, .dmg, DMG, .DMG)